### PR TITLE
Add Samtools bedcov as a process

### PR DESCRIPTION
**Pull request by marcellevstek**
* [x] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [x] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [x] All inputs are used in process.
* [x] All output fields have a value assigned to them.

### CODE REVIEW

File: resolwe_bio/processes/samtools/samtools_bedcov.py [hunk 1]
```diff
@@ -163,51 +133,52 @@ def run(self, inputs, outputs):
         name = f"{Path(inputs.bedfile.output.bed.path).stem}"
         coverage_report = f"{name}_coverage.txt"
```
Comment (marcellevstek) [inline resolwe_bio/processes/samtools/samtools_bedcov.py:134]: Maybe we should save this as a TSV file, as it is indeed tab delimited.

File: resolwe_bio/processes/samtools/samtools_bedcov.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    data_name = "{{ bam|name|default('?') }}"
+    version = "1.0.0"
+    entity = {"type": "sample"}
+    scheduling_class = SchedulingClass.BATCH
+
+    class Input:
+        """Input fields for SamtoolsBedcov."""
+
+        bams = ListField(
+            DataField(data_type="alignment:bam"), label="Input BAM files", required=True
```
Comment (jkokosar) [inline resolwe_bio/processes/samtools/samtools_bedcov.py:87]: Omit `required=True` as it is redundant.

File: resolwe_bio/processes/samtools/samtools_bedcov.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    category = "Samtools"
+    data_name = "{{ bam|name|default('?') }}"
+    version = "1.0.0"
+    entity = {"type": "sample"}
+    scheduling_class = SchedulingClass.BATCH
+
+    class Input:
+        """Input fields for SamtoolsBedcov."""
+
+        bams = ListField(
```
Comment (jkokosar) [inline resolwe_bio/processes/samtools/samtools_bedcov.py:86]: Each BAM from `bams` input is one sample. Defining entity = {"type": "sample"} will result in a new sample being crated, as the inputs are already part of individual/separate samples. 
If the goal is to attach this data to the existing samples, then each BAM should be processed separately (remove ListField). If there is a need for multisample bedcov process, this should be a separate analysis.

File: resolwe_bio/processes/samtools/samtools_bedcov.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+    """
+
+    slug = "samtools-bedcov"
+    process_type = "data:alignment:bam:samtools"
+    name = "Samtools bedcov"
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/genialis/resolwebio/rnaseq:6.0.0"},
```
Comment (romunov) [inline resolwe_bio/processes/samtools/samtools_bedcov.py:70]: Common image should be smaller (faster pull), and it also includes samtools.

File: resolwe_bio/processes/samtools/samtools_bedcov.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        """Input fields for SamtoolsBedcov."""
+
+        bams = ListField(
+            DataField(data_type="alignment:bam"), label="Input BAM files", required=True
+        )
+        bedfile = DataField(
+            data_type="bed",
+            label="Target BED file",
+            description="Target BED file with regions to extract.",
+            required=True,
```
Comment (jkokosar) [inline resolwe_bio/processes/samtools/samtools_bedcov.py:93]: As above, you can omit this.

File: resolwe_bio/processes/samtools/samtools_bedcov.py (hunk 1 above)
Comment (jkokosar) [reply resolwe_bio/processes/samtools/samtools_bedcov.py:134]: I vote to retain .txt

**Comment by marcellevstek** [comment]
What is your opinion on processes handling multiple samples in one analysis? Should we use a separate processes for single and multiple sample, or keep the option to output multiple samples in one file as in this code? Matjaz said that having multiple samples in one output file is not optimal, as most processes in Expressions work on individual samples instead of collections.

File: resolwe_bio/processes/samtools/samtools_bedcov.py (hunk 3 above)
Comment (marcellevstek) [reply resolwe_bio/processes/samtools/samtools_bedcov.py:86]: Ok, we will then use it per sample.

File: resolwe_bio/processes/samtools/samtools_bedcov.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    Input columns : [chr, start, stop, coverage sum]
+    Output columns : [chr, start, stop, coverage mean]
+    """
+    coverage_df = pd.read_csv(
+        StringIO(output),
+        sep="\t",
+        names=["chrom", "start", "end", "coverage"],
+    )
+
+    coverage_df["coverage"] = coverage_df["coverage"] / (
```
Comment (jkokosar) [inline resolwe_bio/processes/samtools/samtools_bedcov.py:37]: You could round the mean coverage to 2 decimal places

File: resolwe_bio/processes/samtools/samtools_bedcov.py [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    process_type = "data:alignment:bam:samtools"
+    name = "Samtools bedcov"
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/s4q6j6e8/resolwebio/common:3.1.0"},
+        },
+        "resources": {
+            "cores": 2,
+            "memory": 16384,
```
Comment (jkokosar) [inline resolwe_bio/processes/samtools/samtools_bedcov.py:70]: I suspect the process to be fairly efficient. We might get away with less resources. I suggest we drop this to 8GB RAM.

File: resolwe_bio/tests/processes/samtool_bedcov_temp.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
+skip_reads = StringField(
```
Comment (jkokosar) [inline resolwe_bio/tests/processes/samtool_bedcov_temp.py:1]: This is here probably by mistake?

**Review by jkokosar — COMMENTED** [review]
Almost ready!

File: resolwe_bio/tests/processes/samtool_bedcov_temp.py (hunk 8 above)
Comment (marcellevstek) [reply resolwe_bio/tests/processes/samtool_bedcov_temp.py:1]: Yes, got included by mistake.

**Review by jkokosar — APPROVED** [review]
Looks good! Please rebase

**Pull request merged**